### DISCUSSION TITLE
Available helper methods to convert matched string to value?

### DISCUSSION THREAD

**Discussion by hrishikesh-k** [unocss/unocss#2654] [Q&A] [answered]
Hello!

I needed a `letter-spacing` utility, but based on this: https://www.unocss.dev/interactive/?s=letter-spacing, it's not available in UnoCSS. So, I wrote my custom rule like:

```ts
import {defineConfig, toEscapedSelector} from 'unocss'
import presetAttributify from '@unocss/preset-attributify'
import presetUno from '@unocss/preset-uno'
export default defineConfig({
  presets: [
    presetAttributify({
      nonValuedAttribute: false,
      prefix: 'u-',
      prefixedOnly: true,
      strict: true
    }),
    presetUno({
      dark: 'class',
      variablePrefix: 'u-'
    })
  ],
  rules: [[
    /^letter-spacing(.+)$/,
    (matchedRule, generator) => {
    return `
      ${toEscapedSelector(generator.rawSelector)} {
        letter-spacing: ${parseFloat(matchedRule[1]!.slice(1)) / 4}rem;
      }
    `
    }
  ]]
})
```

However, as you can see, this is only handling the case where `u-letter-spacing="4"` (`4` is an example value, but it could be any +ve or -ve number). Since UnoCSS can also handle other values like fractions, or values in `[]`, etc. I wonder if there are any helper methods available in UnoCSS to add support for this. If not, I'd have to do it all manually, but since UnoCSS does it internally for other rules, I believe me writing an extra piece of code would not just be redundant, but could also go out-of-sync if something changes with UnoCSS. So to maintain consistency with the rest of the rules, I'd like to know if any kind of value can be supported by custom rules.

**Comment by zyyv** [comment, marked as answer]
Sorry reply to you so late. You can check this https://unocss.dev/interactive/?s=tracking-
If you don't use some rules in UnoCSS, you can search it in Tailwind or Windi, we are fully compatible with them.
https://tailwindcss.com/docs/letter-spacing#setting-the-letter-spacing

**Reply by hrishikesh-k** [reply]
Oh that's good to know, thanks!
